scripts/src/test/org/oppia/android/scripts/XMLSyntaxCheckTest.kt

```diff
@@ -0,0 +1,155 @@
+package org.oppia.android.scripts
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.PrintStream
+
+/** Tests for [XMLSyntaxCheck]. */
+class XMLSyntaxCheckTest {
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = java.lang.System.out
+
+  @Rule
+  @JvmField
+  public var tempFolder: TemporaryFolder = TemporaryFolder()
+
+  @Before
+  fun setUpTests() {
+    tempFolder.newFolder("testfiles")
+    java.lang.System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    java.lang.System.setOut(originalOut)
+  }
+
+  @Test
+  fun testXmlSyntax_validXML_xmlSyntaxIsCorrect() {
+    val validXML =
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+        "<shape xmlns:android=\"http://schemas.android.com/apk/res/android\"\n" +
+        "  android:shape=\"rectangle\">\n" +
+        "  <solid android:color=\"#3333334D\" />\n" +
+        "  <size android:height=\"1dp\" />\n" +
+        "</shape>"
+
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(validXML)
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(
+      ScriptResultConstants.XML_SYNTAX_CHECK_PASSED
+    )
+  }
+
+  @Test
+  fun testXmlSyntax_invalidOpeningTag_xmlSyntaxIsInCorrect() {
+    val invalidXML =
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+        "<shape xmlns:android=\"http://schemas.android.com/apk/res/android\"\n" +
+        "  android:shape=\"rectangle\">\n" +
+        "  <<solid android:color=\"#3333334D\" />\n" +
+        "  <size android:height=\"1dp\" />\n" +
+        "</shape>"
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(invalidXML)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(
+      ScriptResultConstants.XML_SYNTAX_CHECK_FAILED
+    )
+    assertThat(outContent.toString().trim()).isEqualTo(
+      "XML syntax error: The content of elements must consist of well-formed" +
+        " character data or markup.\n" +
+        "lineNumber: 4\n" +
+        "columnNumber: 4\n" +
+        "File: [ROOT]/testfiles/TestFile.xml"
```

Comment:
I'm starting to change my thinking on this. Instead of ``//`` maybe we should actually just fully qualify it (e.g. absolute path).

For the tests, you can just use multiple ``contains`` checks instead of validating the exact output, and the contains can just verify ``testfiles/TestFile.xml`` is present. Ditto for other tests & #3340.

WDYT?

Reply:
SGTM. DONE.